homeassistant/helpers/helper_config_entry_flow.py

```diff
@@ -38,19 +44,28 @@ async def async_step(self, _user_input: dict[str, Any] | None = None) -> FlowRes
             except vol.Invalid as exc:
                 errors["base"] = str(exc)
             else:
+                self._options.update(user_input)
```

Comment:
Shouldn't this only be done when you're also going to update the actual entry (so inside the if-statement)

Reply:
If we do that, we can't collect options from several steps.
For example the group type in the group flow from the first config step wouldn't be saved.